src/static/css/page.css

```diff
@@ -295,6 +296,28 @@ img {
   max-width: 100%;
   height: auto;
 }
+
+.anchor-link {
+  text-decoration: none;
+  padding-left: 30px;
+  margin-left: -30px;
+}
+
+figure .anchor-link {
+  text-decoration: underline;
+}
+
+.anchor-link:hover::before,
+.anchor-link:focus::before {
+  content: url('/static/images/link.svg');
+  content: url('/static/images/link.svg') / '';
+  left: -30px;
+  position: relative;
+  float: left;
```

Comment:
Does float apply here?

Reply:
Yes without this it reserves some space evem with the `width: 0` after this line - to be honest, I'm not sure why, but float fixes it:

![image](https://user-images.githubusercontent.com/10931297/79670740-1a674900-81bd-11ea-88e3-8ab73c6d5426.png)

Reply:
Ah figured it out!

The `::before` pseudo-element is an inline element by default, and I stupidly forgot the `width` is applied to the whole pseudo-element and not the `content` (which is an image). So, without setting `float: left;`, the `width` is ignored (as element is inline) and hence the extra spacing.

So we either need `float: left;` to implicitly covert into a block element or I can set `display: inline-block;` to so similar. The advantage of `float: left;` is that it takes it out of the flow completely, so it is ignored when the link image is slightly larger than the text (like it is for figcaptions), so it doesn't change the height of the link element and cause a slight movement on hover.

So yes you need `float:left;` and `left:-30px;` (which I since changed to `left: -25px;` btw as looks nicer).